java/src/main/java/sh/platform/languages/SampleServlet.java

```diff
@@ -0,0 +1,57 @@
+package sh.platform.languages;
+
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+import static java.util.Optional.ofNullable;
+
+public class SampleServlet extends HttpServlet {
+
+    private static final long serialVersionUID = -3462096228274971485L;
+
+    @Override
+    protected void doGet(HttpServletRequest request, HttpServletResponse response)
+            throws IOException {
+
+        final String pathInfo = ofNullable(request.getPathInfo()).orElse("");
+        switch (pathInfo) {
+            case "/":
+                response.setContentType("application/json");
+                response.setStatus(HttpServletResponse.SC_OK);
+                response.getWriter().println(SamplesAvailable.getOptions());
+                return;
```

Comment:
If I'm reading this right, the / route will return... a JSON blob?  Why?  We don't use that anywhere, do we?  The / route should be an HTML page with everything in it.  

The output paths also seem to be missing.

There's also no semaphore locking around the sample code executions that I can see.  Am I missing it?

Reply:
> If I'm reading this right, the / route will return... a JSON blob? Why? We don't use that anywhere, do we? 
On Python on the root, we're returning the list of services:

```python
@app.route('/python/')
def root():
    return create_list()
```

> The / route should be an HTML page with everything in it.
On python, it's returning text/plain.
```
resp.headers['Content-Type'] = 'text/plain'
```
I believe just on PHP we're returning as HTML because I believe on PHP it fine.

> The output paths also seem to be missing.
I put the services that we're supporting, for now, anything else will return 404.

> There's also no semaphore locking around the sample code executions that I can see. Am I missing it?
The Jetty server will handle it for us.